mesonbuild/mdevenv.py

```diff
@@ -21,12 +23,17 @@ def add_arguments(parser: argparse.ArgumentParser) -> None:
     parser.add_argument('command', nargs=argparse.REMAINDER,
                         help='Command to run in developer environment (default: interactive shell)')
 
-def get_windows_shell() -> str:
+def get_windows_shell() -> T.Optional[str]:
     mesonbuild = Path(__file__).parent
     script = mesonbuild / 'scripts' / 'cmd_or_ps.ps1'
-    command = ['powershell.exe', '-noprofile', '-executionpolicy', 'bypass', '-file', str(script)]
-    result = subprocess.check_output(command)
-    return result.decode().strip()
+    for shell in POWERSHELL_EXES:
+        try:
+            command = [shell, '-noprofile', '-executionpolicy', 'bypass', '-file', str(script)]
+            result = subprocess.check_output(command)
+            return result.decode().strip()
+        except subprocess.CalledProcessError:
```

Comment:
subprocess.Popen is allowed to raise OSError in addition to subprocess.CalledProcessError -- in this case specifically, it is reasonably likely that FileNotFoundError may be raised, for example if pwsh.exe does not exist.

We should catch this too.

Reply:
Makes a lot of sense, yes. Updated.